### DISCUSSION TITLE
ShouldDownloadAndExtractFirefoxLinuxBinary on Windows

### DISCUSSION THREAD

**Discussion by jnyrup** [hardkoded/puppeteer-sharp#2130] [General]
`BrowserFetcher.ExtractTar` assumes that one has a tool `tar` available capable of decompressing `.tar.bz2` archives.
I think that's a valid assumption on Linux systems.

On my Windows machine the native `tar` is not able to, so `ShouldDownloadAndExtractFirefoxLinuxBinary` consistently fails on my machine.
It's not a problem per se, as I now know why it fails.

Since the CI also runs on Linux machines, one idea is to skip that test on Windows machines, like `PublishSingleFileTests.ShouldWork` is skipped on non-windows machines.

What do you think about that?

**Comment by kblok** [comment]
Yeah. We have a `SkipNonWindowsFact` attribute. We could implement a `SkipOnWindowsFact`.

(sorry for the late reply)

**Comment by jnyrup** [comment]
Just for the record, the exact error from my `tar` command is
```
tar: Error opening archive: Can't initialize filter; unable to run program "bzip2 -d"
```

We could consider improving the experience by checking `Process.ExitCode` after invoking `Process.WaitForExit()` to see if the process succeed.
